src/server/pkg/storage/fileset/util.go

```diff
@@ -108,3 +122,84 @@ func DirUpperBound(p string) string {
 	}
 	return strings.TrimRight(p, "/") + "0"
 }
+
+// Iterator provides functionality for imperative iteration over a file set.
```

Comment:
This needs to be better documented if it's exposed. How do I use this? When is iteration done? , if I don't close it, does it leak?  A code snippet with a for loop showing typical use is probably the shortest way to document it.